tmate-ssh-server.c

```diff
@@ -473,7 +495,28 @@ void tmate_ssh_server_main(struct tmate_session *session, const char *keys_dir,
 	if (!client->session)
 		tmate_fatal("Cannot initialize session");
 
+	fds[0].fd = ssh_bind_get_fd(daemon_bind);
+	fds[0].events = POLLIN;
+	fds[1].fd =  ssh_bind_get_fd(client_bind);
+	fds[1].events = POLLIN;
+
 	for (;;) {
+		ret = poll(fds, 2, 10 * 1000);
+
+		if (ret == 0 || ret < 0 && errno == EINTR)
+			continue;
+
+		if (fds[0].revents & POLLIN) {
+			bind = daemon_bind;
+			handle_role_daemon = true;
+			handle_role_pty_client = false;
+		}
+		else if (fds[1].revents & POLLIN) {
+			bind = client_bind;
+			handle_role_daemon = false;
+			handle_role_pty_client = true;
+		}
```

Comment:
I think you should also test at least for `POLLERR` (or more generally: `if (fds[1].revents & ~POLLIN != 0)`) and just fail the whole program on error (`tmate_fatal`).